timesketch/frontend-ng/src/views/Sketch.vue

```diff
@@ -14,341 +14,345 @@ See the License for the specific language governing permissions and
+    <div v-if="sketch.id" style="height: 30%">
+      <!-- Empty state -->
+      <v-container v-if="!hasTimelines && !loadingSketch" fill-height fluid>
+        <v-row align="center" justify="center">
+          <v-sheet class="pa-4">
+            <center>
+              <div style="font-size: 2em" class="mb-3">It's empty around here</div>
+              <ts-upload-timeline-form btn-size="normal" btn-type="outlined"></ts-upload-timeline-form>
+            </center>
+          </v-sheet>
+        </v-row>
+      </v-container>
 
-    <!-- Archived state -->
-    <v-container v-if="isArchived && !loadingSketch" fill-height fluid>
-      <v-row align="center" justify="center">
-        <v-sheet class="pa-4">
-          <center>
-            <div style="font-size: 2em" class="mb-3">This sketch is archived</div>
-            <v-btn outlined color="primary" @click="unArchiveSketch()"> Bring it back </v-btn>
-          </center>
-        </v-sheet>
-      </v-row>
-    </v-container>
+      <!-- Archived state -->
+      <v-container v-if="isArchived && !loadingSketch" fill-height fluid>
+        <v-row align="center" justify="center">
+          <v-sheet class="pa-4">
+            <center>
+              <div style="font-size: 2em" class="mb-3">This sketch is archived</div>
+              <v-btn outlined color="primary" @click="unArchiveSketch()"> Bring it back </v-btn>
+            </center>
+          </v-sheet>
+        </v-row>
+      </v-container>
 
-    <!-- Rename sketch dialog -->
-    <v-dialog v-model="renameSketchDialog" width="600">
-      <v-card class="pa-4">
-        <ts-rename-sketch @close="renameSketchDialog = false"></ts-rename-sketch>
-      </v-card>
-    </v-dialog>
+      <!-- Rename sketch dialog -->
+      <v-dialog v-model="renameSketchDialog" width="600">
+        <v-card class="pa-4">
+          <ts-rename-sketch @close="renameSketchDialog = false"></ts-rename-sketch>
+        </v-card>
+      </v-dialog>
 
-    <!-- Top horizontal toolbar -->
-    <v-app-bar
-      v-if="!loadingSketch"
-      app
-      clipped-left
-      flat
-      :color="$vuetify.theme.dark ? '#121212' : 'white'"
-      :style="[
-        $vuetify.theme.dark
-          ? { 'border-bottom': '1px solid hsla(0,0%,100%,.12) !important' }
-          : { 'border-bottom': '1px solid rgba(0,0,0,.12) !important' },
-      ]"
-    >
-      <v-btn icon @click="toggleLeftPanel">
-        <v-icon>mdi-menu</v-icon>
-      </v-btn>
+      <!-- Top horizontal toolbar -->
+      <v-app-bar
+        v-if="!loadingSketch"
+        app
+        clipped-left
+        flat
+        :color="$vuetify.theme.dark ? '#121212' : 'white'"
+        :style="[
+          $vuetify.theme.dark
+            ? { 'border-bottom': '1px solid hsla(0,0%,100%,.12) !important' }
+            : { 'border-bottom': '1px solid rgba(0,0,0,.12) !important' },
+        ]"
+      >
+        <v-btn icon @click="toggleLeftPanel">
+          <v-icon>mdi-menu</v-icon>
+        </v-btn>
```

Comment:
UX nit: on an empty sketch this icon still exists but will not toggle anything since there is no left panel yet. I would recommend to hide this on a sketch without any timeline.

Reply:
Good idea. and Ack, will do that in a followup PR as this is related to some other aspects of the left side panel.